hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -163,6 +170,143 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     }
   }
 
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace);
+    Preconditions.checkArgument(namespace.levels().length == 1,
+        "Cannot support multi part namespace in Hive MetaStore: %s", namespace);
+    try {
+      clients.run(client -> {
+        client.createDatabase(convertToDatabase(namespace, meta));
+        return null;
+      });
+
+    } catch (AlreadyExistsException e) {
+      throw new org.apache.iceberg.exceptions.AlreadyExistsException("Namespace '%s' already exists!",
+            namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to create namespace " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to createDatabase(name) " + namespace.toString() + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    if (namespace.levels().length > 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      return namespace.isEmpty() ? clients.run(HiveMetaStoreClient::getAllDatabases)
+          .stream().map(name -> Namespace.of(name))
+          .collect(Collectors.toList()) : clients.run(client -> client.getDatabases(namespace.level(0)))
```

Comment:
What is the semantics here? IIUC, Hive only supports one layer of namespace (db), so what is the semantics of db under db? 

Also here `getDatabases(xxx)` is more like getting DBs with name pattern, not DBs under db. So here the question is are we going to support `listNamespaces` under namespace for Hive?